Level Rankings on play page

edinstance opened the pull request:
This is a pr for #124, it includes:

- Renaming a leaderboard query
- Creating a new leaderboard query for getting rankings per level
- Displaying the leaderboard data for each level on the play page

This has now been edited to combine the two leaderboard queries and have an optional level that can be passed in.

delasy commented on convex/leaderboard.ts:
```diff
@@ -22,8 +22,20 @@ export const getGlobalRankings = query({
+    if (args.level) {
+      res = await db
+        .query("levelRankings")
+        .filter((q) => q.eq(q.field("level"), args.level))
```
you should use `withIndex` in this case

edinstance replied:
Fixed in 6bbeae44402ccec7cfc35a4aa5a0b6324d781fd6

edinstance commented:
Updated this pr to use an index when fetching the leaderboard for each level and also added buttons from the mini leaderboards on the watch and play pages that lead to the main leaderboard page.

delasy reviewed (approved):
lgtm
